resolwe/flow/views.py

```diff
@@ -451,10 +452,16 @@ class CollectionViewSet(ResolweCreateModelMixin,
                         viewsets.GenericViewSet):
     """API view for :class:`Collection` objects."""
 
-    queryset = Collection.objects.all().prefetch_related('descriptor_schema', 'contributor')
+    queryset = Collection.objects.all().prefetch_related(
+        'descriptor_schema',
+        'contributor',
+        Prefetch('data', queryset=Data.objects.all().order_by('created'))
```

Comment:
Created is not OK, we should order by ID. Objects may be created within the same microsecond and in this cases the order among them will be random.